Enable the `no-var` linting rule in more core files

timvandermeij opened the pull request:
This is mostly done using the `gulp lint --fix` command, and manual changes in a small number of cases where ESLint couldn't determine what to use, in an effort to reduce the number of disabled ESLint rules in the core folder and to use modern syntax.

timvandermeij committed:
Enable the `no-var` linting rule in `src/core/metrics.js`

timvandermeij committed:
Enable the `no-var` linting rule in `src/core/unicode.js`

timvandermeij committed:
Enable the `no-var` linting rule in `src/core/ccitt_stream.js`

timvandermeij committed:
Enable the `no-var` linting rule in `src/core/glyphlist.js`

timvandermeij committed:
Enable the `no-var` linting rule in `src/core/primitives.js`

timvandermeij commented:
/botio test

Snuffleupagus commented on src/core/glyphlist.js:
```diff
@@ -13,11 +13,10 @@
  * limitations under the License.
  */
 /* no-babel-preset */
-/* eslint-disable no-var */
```
*Including this here for future reference, to clarify that we've considered the implications of these changes:*

Here, and in `src/core/unicode.js`, we include these files as-is in the "legacy"-builds because of issues with Babel; please note https://github.com/mozilla/pdf.js/blob/052db45e7a26fead95b5233e3425008784daa8cf/gulpfile.js#L198-L200

Hence this PR will result in `let`/`const` being included even in the "legacy"-builds, however that should be fine given the following compatibility information:
 - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/let#browser_compatibility
 - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/const#browser_compatibility

Note also that IE 11 support is now explicitly removed, and this CSS-compatibility configuration for the `GENERIC`-builds: https://github.com/mozilla/pdf.js/blob/052db45e7a26fead95b5233e3425008784daa8cf/gulpfile.js#L78-L90

Snuffleupagus commented on src/core/unicode.js:
```diff
@@ -1629,13 +1628,13 @@ var getNormalizedUnicodes = getArrayLookupTableFactory(function () {
   }
-  var s = "";
-  for (var ii = charsLength - 1; ii >= 0; ii--) {
+  let s = "";
+  for (let ii = charsLength - 1; ii >= 0; ii--) {
     s += chars[ii];
   }
   return s;
```
*Nit:* Can we please change this to use a buffer, rather than string concatenation, since that should be more efficient for short strings (which is what we're dealing with here). 

*Obviously making these changes in a separate commit would be fine.*

timvandermeij replied:
Done in a new commit. Good idea!

Snuffleupagus reviewed (approved):
Looks great, thank you!

I've added one (small) suggestion, but it's obviously not a blocker :-)

timvandermeij committed:
Use a buffer instead of string concatenation in `reverseIfRtl` in `src/core/unicode.js`

This avoids creating intermediate strings and should be slightly more
efficient.

timvandermeij commented:
/botio unittest
